src/pages/CoachProfile/GenerateToken/TokenGenerator.tsx

```diff
@@ -0,0 +1,67 @@
+import { Box, IconButton, InputAdornment, TextField } from '@mui/material';
+import React, { useState } from 'react';
+import { Button } from '../../../components/Button';
+import axios, { AxiosError } from 'axios';
+import { getCookie } from 'typescript-cookie';
+import clipboardCopy from 'clipboard-copy';
+import { Add, ContentCopy } from '@mui/icons-material';
+import { ContainerStyles } from './TokenGenerator.styles';
+
+type TokenGeneratorProps = {
+  showSnackbar: (message: string, severity: 'success' | 'error') => void;
+};
+
+export const TokenGenerator = ({ showSnackbar }: TokenGeneratorProps) => {
+  const [token, setToken] = useState('');
+
+  const generateToken = async () => {
+    try {
+      const response = await axios.get('http://localhost:8081/api/user/token/generate', {
+        headers: { Authorization: `Bearer ${getCookie('userToken')}` },
+      });
+      if (response.status === 200) {
+        setToken(response.data);
+      }
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        showSnackbar(error.message, 'error');
+      }
+    }
+  };
+
+  const handleCopy = () => {
+    clipboardCopy(token)
+      .then(() => {
+        showSnackbar('Token copied to clipboard', 'success');
+      })
+      .catch(error => {
+        showSnackbar('Failed to copy token to clipboard', 'error');
+      });
+  };
+
+  return (
+    <Box sx={ContainerStyles}>
+      <Button type="button" onClick={generateToken}>
+        <Add sx={{ mr: 1 }} />
+        Generate invitation token
```

Comment:
Może tutaj zastosujmy podejście z dokumentacji MUI? https://mui.com/material-ui/react-button/#buttons-with-icons-and-label